c/version.h

```diff
@@ -33,6 +33,13 @@
 #define FORCEINLINE static inline
 #endif
 
+/* Clang and recent versions of GC provide `__has_builtin` for
+   checking for builtins. This definition completes a typical pattern
+   to make it usable everywhere: */
+#ifndef __has_builtin
+# define __has_builtin(x) 0
+#endif
```

Comment:
I have often seen the recommendation *not* to define your own `__has_builtin` or similar macros since they are part of the implementation, and instead to define and use your own macro name, e.g.:

```suggestion
#ifdef __has_builtin
# define CHEZ_HAS_BUILTIN(x) __has_builtin(x)
#else
# define CHEZ_HAS_BUILTIN(x) 0
#endif
```

This recommendation is important for public headers so that you are not potentially affecting other projects' uses of `__has_builtin`. I don't think version.h gets installed so I'm not sure how important it is in that case. The clang documentation does recommend defining `__has_builtin` if not available, just as you've done.

Reply:
Although these are private headers, I agree that defining a different name in better. Someone reading/modifying "pb.h" or "atomic.h" might spend less time wondering whether the use of `__has_builtin` needs to be guarded with a check to see whether `__has_builtin` exists.

As a minor point, I'd like to go with `C_COMPILER_HAS_BUILTIN`, though, since I can't help reading `CHEZ_HAS_BUILTIN` as a "Chez Scheme has builtin". Other private definitions generally don't use a prefix like `CHEZ`.

Reply:
> I'd like to go with `C_COMPILER_HAS_BUILTIN`, though

As you wish. For definitions in public headers, it would be important to use a macro name that will not conflict with other macros, and the usual way is to include your project name in the macro name. Looking through some of the headers installed on my system, Ruby for example defines `RBIMPL_HAS_BUILTIN`, Eigen defines `EIGEN_HAS_BUILTIN`, and xxHash defines `XXH_HAS_BUILTIN`. If this is a private header it's only a problem if you're including a header from another project and they have inadvisedly defined a macro `C_COMPILER_HAS_BUILTIN`.